src/data_gradients/datasets/detection/yolo_format_detection_dataset.py

```diff
@@ -135,25 +129,11 @@ def load_labels(self, index: int) -> np.ndarray:
                 try:
                     labels.append(list(map(float, lines_elements)))
                 except ValueError as e:
-                    raise ValueError(
-                        f"Invalid label: {line} from {label_path}.\nExpected 5 elements (class_id, cx, cy, w, h), got {len(lines_elements)}."
-                    ) from e
+                    raise ValueError(f"Invalid label: {line} from {path}.\nExpected 5 elements (class_id, cx, cy, w, h), got {len(lines_elements)}.") from e
```

Comment:
I think this is brittle design. Some YOLO annotation files also contains keypoint annotations in the same txt file next to bounding boxes. This code would crash on such files. 

Can we extract standalone "read_yolo_annotation(lines: List[str], on_parse_error='raise')" method that we can convert with unit tests to ensure it:

* Can handle spaces, tabs and other whitespace symbols without crashing
* Would skip empty line or commented lines (Starting with `#`). Yes, it is a thing in some YOLO annotations as well
* Check that indeed the first element is a label (it's an integer and not float)
* If `on_parse_error='raise'` then it would raise is unexpected string comes in and if `on_parse_error='warining'` it would emit a warning and skip this line.

Reply:
What do you think about this ?
98af9d2d44a5a4616c2e095793f35565a9436117

Reply:
Good!